From b6a6e694be63796f15fe45c79e2f70840514eefe Mon Sep 17 00:00:00 2001
From: jayanth1 <shankar.srinivas.jayanthi@intel.com>
Date: Thu, 5 Nov 2020 18:57:04 +0530
Subject: [PATCH] Batteryless behavior changes for CIV

when Android does not detect a battery device for CIV mode
battery-related default values are required to update
according to the Android 9 release or above

Change-Id: Ic7c48c0e983e65421ecdc2d913fa7d1fc1e99ba0
Tracked-On: OAM-94516
Signed-off-by: jayanth1 <shankar.srinivas.jayanthi@intel.com>

diff --git a/health/HealthService.cpp b/health/HealthService.cpp
index 62c8048..b6ff33d 100644
--- a/health/HealthService.cpp
+++ b/health/HealthService.cpp
@@ -75,9 +75,9 @@ int healthd_board_battery_update(struct android::BatteryProperties *props)
     if (get_battery_properties(props))
         platformPowerSupplyType = BATTERY;
     if (platformPowerSupplyType == CONSTANT_POWER) {
-        props->batteryStatus = android::BATTERY_STATUS_FULL;
-        props->batteryHealth = android::BATTERY_HEALTH_GOOD;
-        props->batteryLevel = 100;
+        props->batteryStatus = android::BATTERY_STATUS_UNKNOWN;
+        props->batteryHealth = android::BATTERY_HEALTH_UNKNOWN;
+        props->batteryLevel = 0;
         props->batteryChargeCounter= 1000000;
         props->batteryCurrent= 1000000;
         props->chargerAcOnline = true;
@@ -85,7 +85,7 @@ int healthd_board_battery_update(struct android::BatteryProperties *props)
         props->chargerWirelessOnline = false;
         props->maxChargingCurrent= 2500000;
         props->maxChargingVoltage= 4300000;
-        props->batteryPresent= true;
+        props->batteryPresent= false;
         props->batteryVoltage= 1200000;
         props->batteryTemperature= 25;
         props->batteryFullCharge= 4200000;
--
2.29.2

